### PR TITLE
feat: remove trailing slash from http request if it exists in url

### PR DESCRIPTION
**Pull request by yquansah**
This PR will remove the trailing slash from the URL of the http request if it exists.

Completes FLI-339

### CODE REVIEW

**Review by yquansah — COMMENTED** [review]
I am very undecided how I wanted to architect the integration tests with this PR. Suggestions would be appreciated 🤔 

I threaded through the `fliptAddr` and constructed my own http client for this purpose.

File: internal/cmd/http.go [hunk 1]
```diff
@@ -99,6 +101,20 @@ func NewHTTPServer(
...
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if strings.HasSuffix(r.URL.Path, "/") {
+				// Panic if URL can not be parsed if a trailing slash is trimmed.
+				nurl, err := url.Parse(strings.TrimSuffix(r.URL.String(), "/"))
+				if err != nil {
+					panic(err)
+				}
+
+				r.URL = nurl
+			}
```
Comment (GeorgeMac) [inline internal/cmd/http.go:114]: `strings.TrimSuffix` actually already calls `strings.HasSuffix`.
I think you can boil this all down to just the following (avoids panic too):
```suggestion
			r.URL.Path = strings.TrimSuffix(r.URL.Path)
```

**Review by GeorgeMac — CHANGES_REQUESTED** [review]
Yeah I think we should consider not putting these in the `API` tests.
They seem like an awkward place to shoe-horn them in.

I think perhaps we're reaching the need for a general light refactoring of test layouts here.
But maybe not just for this issue. I will ruminate on that a bit.

For now though,  maybe this is something we can just unit test? My thinking here is: what is most important about this issue is that all of our **clients work**. Not that we do or do not support trailing slashes. We need a single working path for all our RPCs at a minimum.

Currently, all our clients work without this feature. Adding it is a nice to have to be more flexible for future clients. When it comes to integration tests that are valuable here: I wonder if what is more impotant as an integration test, is to run our clients in different languages against Flipt. Instead of trying to fabricate this implementation detail in an integration test, we want an easy way to slot in a new client and ensure it works against a running Flipt.

What do you think too @markphelps ?

(Aside: maybe these integration tests should be renamed compatability tests)

File: internal/cmd/http.go (hunk 1 above)
Comment (yquansah) [reply internal/cmd/http.go]: @GeorgeMac Thank you George!

**Comment by markphelps** [comment]
> Yeah I think we should consider not putting these in the `API` tests. They seem like an awkward place to shoe-horn them in.
> 
> I think perhaps we're reaching the need for a general light refactoring of test layouts here. But maybe not just for this issue. I will ruminate on that a bit.
> 
> For now though, maybe this is something we can just unit test? My thinking here is: what is most important about this issue is that all of our **clients work**. Not that we do or do not support trailing slashes. We need a single working path for all our RPCs at a minimum.
> 
> Currently, all our clients work without this feature. Adding it is a nice to have to be more flexible for future clients. When it comes to integration tests that are valuable here: I wonder if what is more impotant as an integration test, is to run our clients in different languages against Flipt. Instead of trying to fabricate this implementation detail in an integration test, we want an easy way to slot in a new client and ensure it works against a running Flipt.
> 
> What do you think too @markphelps ?
> 
> (Aside: maybe these integration tests should be renamed compatability tests)

Agreed about running clients in other languages for compatibility testing, adding this as an IT prob doesn't fit within out existing IT stack

**Review by markphelps — APPROVED** [review]

**Pull request merged**